src/main/java/taskList/TaskList.java

```diff
@@ -0,0 +1,211 @@
+     * add task functionand then returns the String to be printed to the user. It throws a
+     * DukeException when an error occurred while adding a task to the list.
+     *
+     * @param instruction the command given by the user
+     * @param item description of the task
+     * @return String to be printed to the user
+     * @throws DukeException
+     */
+    public String addItem(String instruction, String item) throws DukeException {
+        if (item.equals("") || item.equals(" ")) {
+            throw new DukeException("Oops! The description cannot be empty >.<");
+        }
+        String output = "";
+        if (instruction.equals("todo")) {
+            output += addTodoItem(item, false);
+        } else if (instruction.equals("deadline")) {
+            output += addDeadline(item, false);
+        } else {
+            output += addEvent(item, false);
+        }
+        output += "\n  New todo item added to the list!";
+        output += "\n  There are now " + (numberOfTasks + 1) + " todo items in the list";
+        return output;
+    }
+
+    /**
+     * It creates a new todo item based on the prams given and returns the String
+     * to be printed to the user.
+     *
+     * @param item description of the todo item
+     * @param isCompleted whether the todo is completed
+     * @return String to be printed to the user
+     */
+    public String addTodoItem(String item, boolean isCompleted) {
+        Todo newTask = new Todo(item, isCompleted);
+        todoList.add(newTask);
+        numberOfTasks += 1;
+        return "  " + newTask.getItem();
+    }
+
+
+    /**
+     * It creates a new Deadline item based on the params given and returns the String
+     * to be printed. It throws a DukeException when an error occurs while adding the
+     * task to the list.
+     *
+     * @param item the description of the Deadline
+     * @param isCompleted whether the Deadline is completed
+     * @return String to be printed to the user
+     * @throws DukeException
+     */
+    public String addDeadline(String item, boolean isCompleted) throws DukeException {
+        Deadline newTask = Parser.parseDeadline(item, isCompleted);
+        todoList.add(newTask);
+        numberOfTasks += 1;
+        return "  " + newTask.getItem();
+    }
+
+    /**
+     * It creates a new Event item based on the params given and returns the String
+     * to be printed. It throws a DukeException when an error occurs while adding the
+     * task to the list.
+     *
+     * @param item description of the Event
+     * @param isCompleted whether the Event is completed
+     * @return String to be printed to the user
+     * @throws DukeException
+     */
+    public String addEvent(String item, boolean isCompleted) throws DukeException {
+        Event newTask = Parser.parseEvent(item, isCompleted);
+        todoList.add(newTask);
+        numberOfTasks += 1;
+        return "  " + newTask.getItem();
+    }
+
```

Comment:
Generally I feel that addDeadline/addEvent can be compressed into a single method addTask as Event/Deadline extends Tasks so it will work.